react-native-purchases-ui/src/index.tsx

```diff
@@ -18,11 +28,36 @@ export function presentPaywallIfNeeded(requiredEntitlementIdentifier: String) {
   RNPaywalls.presentPaywallIfNeeded(requiredEntitlementIdentifier);
 }
 
-const ComponentName = 'Paywall';
-
-export const Paywall =
-  UIManager.getViewManagerConfig(ComponentName) != null
-    ? requireNativeComponent(ComponentName)
+const InternalPaywall =
+  UIManager.getViewManagerConfig('Paywall') != null
+    ? requireNativeComponent('Paywall')
     : () => {
       throw new Error(LINKING_ERROR);
     };
+
+export const Paywall: React.FC<FlexPaywallFooterViewProps> = ({style, children}) => (
+  <InternalPaywall style={[{flex: 1}, style]}/>
+);
+
+const InternalPaywallFooterView = UIManager.getViewManagerConfig('Paywall') != null
+  ? requireNativeComponent('RCPaywallFooterView')
+  : () => {
+    throw new Error(LINKING_ERROR);
+  };
+
+type FlexPaywallFooterViewProps = {
+  style?: StyleProp<ViewStyle>;
+  children?: ReactNode;
+};
+
+export const PaywallFooterContainerView: React.FC<FlexPaywallFooterViewProps> = ({style, children}) => (
```

Comment:
This can go separately, but we should add these components to the API tests if possible

Reply:
hmm, right, we don't have API tests at all for paywalls, so I am going to set that up in another PR